dotCMS/src/integration-test/java/com/dotmarketing/portlets/categories/business/CategoryFactoryTest.java

```diff
@@ -0,0 +1,273 @@
+        assertTrue(categoryFactory.getChildren(parent).isEmpty());
+        assertTrue(categoryFactory.findChildrenByFilter(parent.getInode(), child.getCategoryName(),null).isEmpty());
+    }
+
+    /**
+     * Method to test: {@link CategoryFactory#getParents(Categorizable, String)}
+     * Given Scenario: create a parent-child category hierarchy the test the  child finders
+     * ExpectedResult: Once removed the finders should return empty and vice-versa
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void Test_Find_Parent_Then_Remove_Em() throws DotDataException {
+        final Category parent = newCategory();
+        assertNotNull(parent);
+        final Category child = newCategory();
+        assertNotNull(child);
+        categoryFactory.addChild(parent, child, null);
+        assertFalse(categoryFactory.getParents(child).isEmpty());
+        categoryFactory.delete(parent);
+        assertTrue(categoryFactory.getParents(child).isEmpty());
+    }
+
+    /**
+     * Method to test: {@link CategoryFactory#findTopLevelCategories()}
+     * Given Scenario: create parent children items call top level finder
+     * ExpectedResult: verify top level is found then delete parent call finder again. expect empty collection
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void Test_Find_Top_Level_Categories() throws DotDataException {
+        final Category parent = newCategory();
+        final Category child = newCategory();
+        categoryFactory.addChild(parent, child, null);
+        assertTrue(categoryFactory.findTopLevelCategories().stream().anyMatch(category -> parent.getInode().equals(category.getInode())));
+        assertTrue(categoryFactory.findTopLevelCategories().stream().noneMatch(category -> child.getInode().equals(category.getInode())));
+    }
+
+    /**
+     * Method to test: {@link CategoryFactory#sortChildren(String)}
+     * Given Scenario: create parent children items alter order, save then find
+     * ExpectedResult: items should be in natural ascending order
+     * @throws DotDataException
+     */
+    @Test
+    public void Test_ReOrder_Categories() throws DotDataException {
+        final Category parent = newCategory();
+        assertNotNull(parent);
+        final Category child1 = newCategory();
+        final Category child2 = newCategory();
+        final Category child3 = newCategory();
+
+        child1.setSortOrder(3);
+        categoryFactory.save(child1);
+        child2.setSortOrder(2);
+        categoryFactory.save(child2);
+        child3.setSortOrder(1);
+        categoryFactory.save(child3);
+
+        categoryFactory.sortChildren(parent.getInode());
+
+        final List<Category> children = categoryFactory.getChildren(parent);
+        int orderCount = 1;
+        for(Category category:children){
+            assertEquals(orderCount++,category.getSortOrder().intValue());
```

Comment:
I think you can also validate against the category inode

Reply:
done